common/src/main/java/com/wynntils/wynn/model/questbook/QuestInfo.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model.questbook;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.mc.utils.ItemUtils;
+import com.wynntils.utils.Pair;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.world.item.ItemStack;
+
+public class QuestInfo {
+    private static final Pattern QUEST_NAME_MATCHER = Pattern.compile("^§.§l([^֎À]*)[֎À]+ $");
+    private static final Pattern STATUS_MATCHER = Pattern.compile("^§.(.*)(?:\\.\\.\\.|!)$");
+    private static final Pattern LENGTH_MATCHER = Pattern.compile("^§a-§r§7 Length: §r§f(.*)$");
+    private static final Pattern LEVEL_MATCHER = Pattern.compile("^§..§r§7 Combat Lv. Min: §r§f(\\d+)$");
+    private static final Pattern REQ_MATCHER = Pattern.compile("^§..§r§7 (.*) Lv. Min: §r§f(\\d+)$");
+
+    private final String name;
+    private final QuestStatus status;
+    private final QuestLength length;
+    private final int level;
+    private final String nextTask;
+    /** Additional requirements as pairs of <"profession name", minLevel> */
+    private final List<Pair<String, Integer>> additionalRequirements;
```

Comment:
Wouldn't this be nicer as it's own record/class?

Reply:
Yeah, probably. If is valid for like 3 quests out of 100 though. I was mad enough at how much effort it took to parse and handle those requirements as it were... 

If anything, we should probably have some higher level representation of requirements ("Fishing lv 100", "Intelligence lv 55", "Combat lv 12") and use it everywhere we're talking about level requirements of some kind of aspect of your character. Until we get that, I'd prefer if this stays like this. Ok?

Reply:
Fine by me